messaging/src/main/java/org/axonframework/commandhandling/gateway/AbstractRetryScheduler.java

```diff
@@ -150,6 +193,42 @@ public B retryExecutor(ScheduledExecutorService retryExecutor) {
             return (B) this;
         }
 
+        /**
+         * Sets the {@link Predicate} used for testing whether failure is transient (returns {@code false}) or not (returns {@code true}).
+         * <p/>
+         * Non-transient failures are never retried.
+         * <p/>
+         * If configured, {@code nonTransientFailures} list is ignored.
+         *
+         * @param nonTransientFailurePredicate a {@link Predicate} (accepting a {@link Throwable} parameter) used for testing transiency of provided failure
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public B nonTransientFailurePredicate(Predicate<Throwable> nonTransientFailurePredicate) {
+            this.nonTransientFailurePredicate = nonTransientFailurePredicate;
+            // noinspection unchecked
+            return (B) this;
+        }
+
+        /**
+         * Sets the list of throwable classes which will be considered as non-transient (including descendant throwable classes).
+         * <p/>
+         * Non-transient failures are never retried.
+         * <p/>
+         * It is ignored if {@code nonTransientFailurePredicate} is configured.
+         * <p/>
+         * By default, it is a list containing only {@code AxonNonTransientException} class.
+         *
+         * @param nonTransientFailures a list of throwable classes used for comparing and testing transiency of provided failure
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public B nonTransientFailures(List<Class<? extends Throwable>> nonTransientFailures) {
+            this.nonTransientFailures.clear();
```

Comment:
Nit: we should check if the provided `nonTransientFailures` is non-null. If we don't, the `addAll` operation will throw a `NullPointerException`.
I'd suggest using the `BuilderUtils#assertNonNull` method for this. Using it will ensure the framework throws an `AxonConfigurationException`.